---
title: "Drug (011 536) high dose injection"
output:
  pdf_document: default
  html_notebook: default
  html_document:
    df_print: paged
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

####Cheng Chen, 12-21-2018


####**Note:** This file only shows the analysis betweeen *011 high dose injection_vs_wt* and *536 high dose injection_vs_wt*. The results between *536 sf3b1 sibling injection_vs_buffer sf3b1 sibling injection* and *536+011 sf3b1 sibling injection_vs_buffer sf3b1 injection*, *536 sf3b1 mutant injection_vs_buffer sf3b1 mutant injection* and *536+011 sf3b1 mutant injection_vs_buffer sf3b1 mutant injection* are not shown because there are not enough significant differential expressed genes between these conditions.


##data preparation
```{r load_packages,echo=FALSE, message=FALSE}
#Load neccessary packages for analysis.
library(ggplot2) 
library(Cairo) 
library(stringr)
library(VennDiagram)
library(clusterProfiler)
library("org.Dr.eg.db")
```

```{r}
#Set the working directory
setwd("/home/chengchen/data/RNAseq/featurecount")
#dir()
```


```{r}
#open summarize count file, count file was obtatined from htseq-count with data integrataion
#Expression Matrix data preprocessing. Exression matris was acquired with the feature count method since it ran faster with this method. 28 samples are loaded in this analysis, but only 011 and 536 drug injection samples will be analyzed. Three samples are dropped because these samples are outliers according to the PCA plot of 31 samples.

count = read.csv("31samplescountmatrix.txt",sep = "\t", header = TRUE, row.names = 1)
count = count[,-c(14,30,31)]
colnames = colnames(count)
sampleCondition = c(rep("011inject",3),rep("011Rpm",2),rep("51Sfm",2),
                    rep("51Sfs",2),"51Sfm","51Sfs",rep("536inject",2),rep("5Sfm",2),
                    rep("5Sfs",2),"5Sfm","5Sfs",rep("Ctl",3),rep("ctRpm",2),
                    "Sfm","Sfs","Sfm","Sfs")
coldata = data.frame(samples=colnames,condition=sampleCondition)

#colnames(count)=rownames(coldata)
rownames(coldata)=colnames(count)
```


Check the expression matrix from featureCount methods.
```{r, message=FALSE}
head(count)
#all(rownames(coldata)==colnames(count))

#import data for further DESeq2 analyis
library("DESeq2")
ddsMatrix = DESeqDataSetFromMatrix(countData = count,colData = coldata,design=~condition)
#head(coldata)
ddsMatrix$condition =  relevel(ddsMatrix$condition,ref = "Ctl")
```

total number of genes before filtering.
```{r}
nrow(ddsMatrix)
```

total number of genes for further analysis after filtering.
```{r}
dds=ddsMatrix[rowSums(counts(ddsMatrix))>=10,] #how to select the number according to sample numbers?
nrow(dds) #check the filtering change
```

```{r, message=FALSE}
#obtain the DEG results
dds=DESeq(dds)
```


```{r, echo=FALSE}
#define a plot function
plotGO = function(dataset) {
  
  OrgDb <- org.Dr.eg.db
  geneList = as.vector(dataset$log2FoldChange)
  names(geneList) = as.character(dataset$entrezgene)
  gene = na.omit(as.character(dataset$entrezgene))

  filename = deparse(substitute(dataset))
  plot_title = paste(filename,"GO_molecurlar_function_enrichment",sep = "")
  ego = clusterProfiler::enrichGO(gene = gene,OrgDb = OrgDb,ont = "MF",pvalueCutoff = 0.05,qvalueCutoff = 0.05,readable = TRUE)
  head(data.frame(ego)[,-8])
  ego = data.frame(ego)
  #pdf_path = paste("/home/chengchen/data/RNAseq/featurecount/figures/",filename,"_GOenrich.pdf",sep = "")
  #CairoPDF(pdf_path, width = 15, height = 8)
  ggplot(ego)+geom_bar(aes(x=reorder(Description,Count),y=Count,fill=-log10(pvalue)),stat="identity")+coord_flip()+scale_fill_gradient(expression(-log["10"](P.value)),low="red",high = "blue")+ggtitle(plot_title)+theme(text = element_text(size = 20))
  #print(figure)
  #dev.off()
  
  #clusterProfiler::dotplot(ego, showCategory=25)
  #kegg analysis
  #kk = clusterProfiler::enrichKEGG(gene = gene,organism = 'dre',pAdjustMethod = "BH",pvalueCutoff = 0.05,qvalueCutoff = 0.05)
  #head(data.frame(kk)[,-8])
  #filestring2 = paste("./figures/",filename,"_KEGGenrichment.pdf",sep = "")
  #pdf(filestring2)
  #barplot(kk,showCategory = 25)
  #dev.off()
}
```


```{r}
plotKEGG = function(dataset) {
  
  OrgDb <- org.Dr.eg.db
  geneList = as.vector(dataset$log2FoldChange)
  names(geneList) = as.character(dataset$entrezgene)
  gene = na.omit(as.character(dataset$entrezgene))

  #kegg analysis
  filename = deparse(substitute(dataset))
  plot_title = paste(filename,"_KEGG_pathway_enrichment",sep = "")
  kk = clusterProfiler::enrichKEGG(gene = gene,organism = 'dre',pAdjustMethod = "BH",pvalueCutoff = 0.05,qvalueCutoff = 0.05)
  head(data.frame(kk)[,-8])
  kk = data.frame(kk)
  #filestring2 = paste("./figures/",filename,"_KEGGenrichment.pdf",sep = "")
  #pdf(filestring2)
  #barplot(kk, main = '011_inject_kegg_pathway_enrichment',xlab="count")
  ggplot(kk)+geom_bar(aes(x=reorder(Description,Count),y=Count,fill=-log10(pvalue)),stat="identity")+coord_flip()+scale_fill_gradient(expression(-log["10"](P.value)),low="red",high = "blue")+ggtitle(plot_title)
  #dev.off()
}
```





011 high dose injection gene expression change compared to the wt.
```{r, message=FALSE}

#DEG between 011inject and wt
res0 = results(dds,contrast = c("condition","011inject","Ctl")) 
res05 = res0[res0$padj<0.05&!is.na(res0$padj),] #remove rows with padj less than 0.05
#res05ordered = res05[order(res05$padj),]
keep1 = abs(res05$log2FoldChange) > 1 #keep those genes with reasonable fold change
res05_lfc = res05[keep1,]
nrow(res05_lfc)
ids = rownames(res05_lfc)
###get zfin_gene_symbol for ensembl genes
library("biomaRt")
ensembl=useMart("ensembl")
ensembl=useDataset("drerio_gene_ensembl",mart=ensembl)
symbols <- getBM(attributes=c('ensembl_gene_id','entrezgene','zfin_id_symbol', 'wikigene_description','name_1006'),filters='ensembl_gene_id',values=ids, mart=ensembl)
#entrezgene is ENTREZID
symbols=symbols[!duplicated(symbols$ensembl_gene_id),]
res05_lfc = as.data.frame(res05_lfc)
res05_lfc$ensembl_gene_id = rownames(res05_lfc)
res05_lfc = merge(res05_lfc,symbols, by="ensembl_gene_id")
#res05_lfc=res05_lfc[,-7] #remove ensembl_gene_id column
res05_lfc_ordered = res05_lfc[order(res05_lfc$padj),]
head(res05_lfc_ordered)
#res05ordered_lfc$symbol = mapIds(org.Dr.eg.db,keys = row.names(res05ordered_lfc),column = "SYMBOL",keytype = "ENSEMBL", multiVals="first")
#data for VennDiagram analysis
write.csv(res05_lfc_ordered,file="inject011_vs_CtlDEG.csv") 
inject011 = res05_lfc_ordered
```

536 high dose injection gene expression change compared to the wt.
```{r,message=FALSE}
#DEG between 536 control and wt
res1 = results(dds,contrast = c("condition","536inject","Ctl")) 
res05 = res1[res1$padj<0.05&!is.na(res0$padj),] #remove rows with padj less than 0.05
keep1 = abs(res05$log2FoldChange) > 1 #keep those genes with reasonable fold change
res05_lfc = res05[keep1,]
nrow(res05_lfc)
ids = rownames(res05_lfc)
###get zfin_gene_symbol for ensembl genes
library("biomaRt")
ensembl=useMart("ensembl")
ensembl=useDataset("drerio_gene_ensembl",mart=ensembl)
symbols <- getBM(attributes=c('ensembl_gene_id','entrezgene','zfin_id_symbol', 'wikigene_description','name_1006'),filters='ensembl_gene_id',values=ids, mart=ensembl)
#entrezgene is ENTREZID
symbols=symbols[!duplicated(symbols$ensembl_gene_id),]
res05_lfc = as.data.frame(res05_lfc)
res05_lfc$ensembl_gene_id = rownames(res05_lfc)
res05_lfc = merge(res05_lfc,symbols, by="ensembl_gene_id")
#res05_lfc=res05_lfc[,-7] #remove ensembl_gene_id column
res05_lfc_ordered = res05_lfc[order(res05_lfc$padj),]
head(res05_lfc_ordered)
#res05ordered_lfc$symbol = mapIds(org.Dr.eg.db,keys = row.names(res05ordered_lfc),column = "SYMBOL",keytype = "ENSEMBL", multiVals="first")
inject536 = res05_lfc_ordered #data for VennDiagram analysis
write.csv(res05_lfc_ordered,file="inject536_vs_CtlDEG.csv") 
```



```{r,message=FALSE}
####plot VennDiagram and find out subsetsspecific_genes_011=setdiff(rownames(inject011),rownames(inject536))
area1 = length(inject011$ensembl_gene_id)
area2 = length(inject536$ensembl_gene_id)
intersectgenes=intersect(inject011$ensembl_gene_id,inject536$ensembl_gene_id)
specific_genes_011=setdiff(inject011$ensembl_gene_id,inject536$ensembl_gene_id)
specific_genes_536 = setdiff(inject536$ensembl_gene_id,inject011$ensembl_gene_id)
```

###VennDiagram
VennDiagram for two drug administration. The total number of differentially expressed genes for 011 high dose injection is 720, and the total number of differentially expressed genes for 536 high dose injection is 120. The overlapped number of genes between two dataset is 80.
```{r}
grid.newpage()
draw.pairwise.venn(area1 = area1,area2 = area2,cross.area = length(intersectgenes),category = c("inject011","inject536"),lty = rep("blank",2),fill = c("light blue", "pink"), alpha = rep(0.5, 2), cat.pos = c(0, 0), cat.dist = rep(0.025, 2))

```



###Gene ontology and KEGG analysis for 011 and 536 high dose injection

```{r}
inject011_specific = inject011[inject011$ensembl_gene_id %in% specific_genes_011,]
inject536_specific = inject536[inject536$ensembl_gene_id %in% specific_genes_536,]
intersect011_536 = inject011[inject011$ensembl_gene_id %in% intersectgenes,]

```

####Analysis results for 011 high dose specific diffrential expressed genes
011 specific genes
```{r}
genes=inject011_specific$zfin_id_symbol
head(genes[genes != ""],600)
```

####GO molecular function enrichment analysis for 011 inject high dose specific genes
```{r, fig.width=25, fig.height=10}
plotGO(inject011_specific)
```

####KEGG enrichment analysis for 011 high dose injection specific genes
```{r}
plotKEGG(inject011_specific)
```


####Analysis results for 536 high dose specific diffrential expressed genes
536 specific genes ((only show the first 40 genes))
```{r}
genes=inject536_specific$zfin_id_symbol
head(genes[genes != ""],50)
```

####GO molecular function enrichment analysis for 011 inject high dose specific genes
```{r, fig.width=25, fig.height=10}
plotGO(inject536_specific)
```

####KEGG enrichment analysis for 011 high dose injection specific genes
```{r, fig.width=8, fig.height=3}
plotKEGG(inject536_specific)
```


###intersect genes between two datasets
011 specific genes (only show the first 50 genes)
```{r}
genes=intersect011_536$zfin_id_symbol
head(genes[genes != ""],600)
```



##Supplementary
####GO molecular function enrichment analysis for 011 inject high dose
```{r, fig.width=25, fig.height=15}
plotGO(inject011)
```


####KEGG enrichment analysis for 011 high dose injection


(Data from file "inject011_vs_ctlDEG.csv" was submmited to DAVID home for analysis, then result was download for further analysis.)

```{r}
#use ggplot2 to plot KEGG process
setwd("/home/chengchen/data/RNAseq/featurecount")
pathway = read.table("/home/chengchen/data/RNAseq/featurecount/011injectDAVID.tsv",header=T,sep="\t") 
pathway$Term<-str_split_fixed(pathway$Term,":",2)[,2] 
#png_path="/home/chengchen/data/RNAseq/featurecount/KEGG.png" 
#CairoPNG(png_path, width = 5.9, height = 3, units='in', dpi=600) 
ggplot(pathway,aes(x=Fold.Enrichment,y=Term)) + 
  geom_point(aes(size=Count,color=-1*log10(PValue)))+ 
  scale_colour_gradient(low="green",high="red")+ 
  labs( color=expression(-log[10](P.value)), size="Gene number", x="Fold enrichment", y="Pathway name",title="011inject Pathway enrichment")+
  theme_bw()+ 
  theme( axis.text.y = element_text(size = rel(1.3)), axis.title.x = element_text(size=rel(1.3)), axis.title.y = element_blank() ) 
```

011inject JAK-STAT signalling enrichment genes:
epor  irf9  il11a il13ara1
il6st lepa  lepr  pik3r3a


####GO molecular function enrichment analysis for 536 high dose injection
```{r,fig.width=14,fig.height=5}
plotGO(inject536)
```


####KEGG pathway enrichment analysis for 536 high dose injection

(Data from file "inject536_vs_ctlDEG.csv" was submmited to DAVID home for analysis, then result was download for further analysis.)

```{r}
pathway = read.table("/home/chengchen/data/RNAseq/featurecount/inject536DAVID.tsv",header=T,sep="\t") 
pathway$Term<-str_split_fixed(pathway$Term,":",2)[,2] 
#png_path="/home/chengchen/data/RNAseq/featurecount/KEGG.png" 
#CairoPNG(png_path, width = 5.9, height = 3, units='in', dpi=600) 
ggplot(pathway,aes(x=Fold.Enrichment,y=Term)) + 
  geom_point(aes(size=Count,color=-1*log10(PValue)))+ 
  scale_colour_gradient(low="green",high="red")+ 
  labs( color=expression(-log[10](P.value)), size="Gene number", x="Fold enrichment", y="Pathway name",title="536inject Pathway enrichment")+
  theme_bw()+ 
  theme( axis.text.y = element_text(size = rel(1.3)), axis.title.x = element_text(size=rel(1.3)), axis.title.y = element_blank() ) 
```

536inject JAK-STAT signaling pathway related genes:
irf9  il11a il12rb2 il6st

